src/ORM/OrmCache.php

```diff
@@ -90,17 +92,31 @@ public function clear($name = null)
     /**
      * Helper method to get the schema collection.
      *
-     * @param string $connection Connection name to get the schema for
-     * @return false|\Cake\Database\Schema\Collection|\Cake\Database\Schema\CachedCollection
+     * @param string|\Cake\Datasource\ConnectionInterface $connection Connection name to get the schema for or a connection instance
+     * @return \Cake\Database\Schema\Collection|\Cake\Database\Schema\CachedCollection
      */
     public function getSchema($connection)
     {
-        /* @var \Cake\Database\Connection $source */
-        $source = ConnectionManager::get($connection);
+        if (is_string($connection)) {
```

Comment:
@markstory @ADmad wanted it for some reason... Can we in the future find a process that somehow makes sure things aren't done and then removed? :(